src/cli/internal/app/call_executor.go

```diff
@@ -0,0 +1,239 @@
+package app
+
+import (
+	"encoding/hex"
+	"fmt"
+	"guillotine-cli/internal/config"
+	"guillotine-cli/internal/types"
+	"strconv"
+	"strings"
+
+	"github.com/evmts/guillotine/bindings/go/evm"
+	"github.com/evmts/guillotine/bindings/go/primitives"
+)
+
+func ValidateCallParameters(params types.CallParameters) error {
+	if params.CallType == "" {
+		return config.NewInputParamError(config.ErrorCallTypeRequired, "call_type")
+	}
+	
+	if !IsValidAddress(params.Caller) {
+		return config.NewInputParamError(config.ErrorInvalidCallerAddress, "caller")
+	}
+	
+	if params.CallType != config.CallTypeCreate && params.CallType != config.CallTypeCreate2 {
+		if !IsValidAddress(params.Target) {
+			return config.NewInputParamError(config.ErrorInvalidTargetAddress, "target")
+		}
+	}
+	
+	if _, err := strconv.ParseUint(params.GasLimit, 10, 64); err != nil {
+		return config.NewInputParamError(config.ErrorInvalidGasLimit, "gas_limit")
+	}
+	
+	if _, err := strconv.ParseUint(params.Value, 10, 64); err != nil {
+		return config.NewInputParamError(config.ErrorInvalidValue, "value")
+	}
+	
+	if !IsValidHex(params.InputData) {
+		return config.NewInputParamError(config.ErrorInvalidInputData, "input_data")
+	}
+	
+	if (params.CallType == config.CallTypeCreate2) && !IsValidHex(params.Salt) {
+		return config.NewInputParamError(config.ErrorInvalidSalt, "salt")
+	}
+	
+	return nil
+}
+
+func ExecuteCall(params types.CallParameters) (*types.CallExecution, error) {
+	if err := ValidateCallParameters(params); err != nil {
+		return nil, err
+	}
+	
+	vm, err := evm.New()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create EVM: %w", err)
+	}
+	defer vm.Close()
+	
+	caller, err := parseAddress(params.Caller)
+	if err != nil {
+		return nil, fmt.Errorf("invalid caller address: %w", err)
+	}
+	
+	value, err := parseWeiValue(params.Value)
+	if err != nil {
+		return nil, fmt.Errorf("invalid value: %w", err)
+	}
+	
+	gasLimit, err := strconv.ParseUint(params.GasLimit, 10, 64)
+	if err != nil {
+		return nil, fmt.Errorf("invalid gas limit: %w", err)
+	}
+	
+	inputData, err := parseBytes(params.InputData)
+	if err != nil {
+		return nil, fmt.Errorf("invalid input data: %w", err)
+	}
```

Comment:
we should be able to just inline all this and handle a single sdk error then this 30 lines of code turns into 3